commit 342f4e5523d933759f42f2c3f7cc4150b5cc9dd8 (refs/changes/68/3568/3)
Author: Jerry Jelinek <jerry.jelinek@joyent.com>
Date:   2018-03-12T15:10:29+00:00 (1 year, 7 months ago)
    
    OS-6695 bhyve failed to pack nvlist
    Reviewed by: Mike Gerdts <mike.gerdts@joyent.com>
    Reviewed by: John Levon <john.levon@joyent.com>
    Approved by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/usr/src/lib/brand/bhyve/zone/boot.c b/usr/src/lib/brand/bhyve/zone/boot.c
index efbff6a32f..88641d567a 100644
--- a/usr/src/lib/brand/bhyve/zone/boot.c
+++ b/usr/src/lib/brand/bhyve/zone/boot.c
@@ -385,7 +385,7 @@ init_debug(void)
 int
 main(int argc, char **argv)
 {
-	int fd;
+	int fd, err;
 	char *zhargv[ZH_MAXARGS] = {
 		"zhyve",	/* Squats on argv[0] */
 		"-H",		/* vmexit on halt isns */
@@ -393,8 +393,8 @@ main(int argc, char **argv)
 		NULL };
 	int zhargc;
 	nvlist_t *nvl;
-	char *nvbuf;
-	size_t nvbuflen;
+	char *nvbuf = NULL;
+	size_t nvbuflen = 0;
 	char zoneroot[MAXPATHLEN];
 	int zrfd;
 	char *zonename;
@@ -441,8 +441,10 @@ main(int argc, char **argv)
 		nvlist_print(stdout, nvl);
 	}
 
-	if (nvlist_pack(nvl, &nvbuf, &nvbuflen, NV_ENCODE_XDR, 0) != 0) {
-		(void) printf("Error: failed to pack nvlist\n");
+	err = nvlist_pack(nvl, &nvbuf, &nvbuflen, NV_ENCODE_XDR, 0);
+	if (err != 0) {
+		(void) printf("Error: failed to pack nvlist: %s\n",
+		    strerror(err));
 		return (1);
 	}
 
